### ISSUE TITLE
repro: Fails to update dvc.lock when AWS credentials expired during execution

### ISSUE THREAD

**Issue by aschuh-hf** [iterative/dvc#9987]
# Bug Report

<!--
## Issue name

Issue names must follow the pattern `command: description` where the command is the dvc command that you are trying to run. The description should describe the consequence of the bug.

Example: `repro: doesn't detect input changes`
-->

## Description

<!--
-->

Commands such as `dvc status` and `dvc repro` now require valid AWS credentials when using `dvc-s3`. This wasn't the case with DVC 2 I belief. The issue with `dvc repro` requiring those credentials is that this may result in the following error when the credentials expired for a long running repro task.

```
Updating lock file 'dvc.lock'                                                                                                                                                                                                                                                                                                                                                      

ERROR: failed to reproduce 'stage': [Errno 22] Bad Request: An error occurred (400) when calling the HeadObject operation: Bad Request
```

Note that the 'stage' was not being executing because of this error, i.e., the error occurs at the start of a stage when DVC tries to determine whether the stage needs to re-run.

### Reproduce

<!--
Step list of how to reproduce the bug
-->

<!--
Example:

1. dvc init
2. Copy dataset.zip to the directory
3. dvc add dataset.zip
4. dvc run -d dataset.zip -o model ./train.sh
5. modify dataset.zip
6. dvc repro
-->

Run `dvc status` in a DVC project which has a S3 bucket configured as remote without having valid AWS credentials. Same for `dvc repro`.

### Expected

<!--
A clear and concise description of what you expect to happen.
-->

AWS credentials should only be required for operations which must read the remote bucket (`dvc push`, `dvc pull`, `dvc gc` etc).

### Environment information

<!--
This is required to ensure that we can reproduce the bug.
-->

**Output of `dvc doctor`:**

```console
$ dvc doctor
DVC version: 3.19.0 (conda)
---------------------------
Platform: Python 3.10.6 on Linux-3.10.0-1127.8.2.el7.x86_64-x86_64-with-glibc2.17
Subprojects:
        dvc_data = 2.16.1
        dvc_objects = 1.0.1
        dvc_render = 0.5.3
        dvc_task = 0.3.0
        scmrepo = 1.3.1
Supports:
        http (aiohttp = 3.8.5, aiohttp-retry = 2.8.3),
        https (aiohttp = 3.8.5, aiohttp-retry = 2.8.3),
        s3 (s3fs = 2023.6.0, boto3 = 1.26.76)
Config:
        Global: /home/aschuh/.config/dvc
        System: /etc/xdg/dvc
Cache types: hardlink, symlink
Cache directory: xfs on /dev/sda1
Caches: local
Remotes: s3, s3
Workspace directory: xfs on /dev/sda1
Repo: dvc (subdir), git
Repo.site_cache_dir: /var/tmp/dvc/repo/b7a036677a9e20e73a791fb023238555
```

**Additional Information (if any):**

<!--
Please check https://github.com/iterative/dvc/wiki/Debugging-DVC on ways to gather more information regarding the issue.

If applicable, please also provide a `--verbose` output of the command, eg: `dvc add --verbose`.
If the issue is regarding the performance, please attach the profiling information and the benchmark comparisons.
-->

**Comment by pmrowla** [comment]
This is probably the same issue as https://github.com/iterative/dvc/issues/9912 which is fixed in DVC `3.20.0` and later, please try updating to the latest release

**Comment by aschuh-hf** [comment]
I've just upgraded to DVC 3.22.1 using `conda` and still get this error.

```
$ dvc status
ERROR: unexpected error - [Errno 22] Bad Request: An error occurred (400) when calling the HeadObject operation: Bad Request
```

**Comment by dberenbaum** [comment]
@aschuh-hf I'm not able to quickly reproduce it, so if you are able to put together some minimal example, that would help. Is `s3 = s3cache` under the cache section left from an old dvc version? I don't think it's supported anymore. Can you drop it, and does that make a difference?

**Comment by aschuh-hf** [comment]
I just deleted `.dvc/tmp/exps/cache` to fix https://github.com/iterative/dvc/issues/9860 as part of the upgrade to DVC 3.22.1 and also wanted to look into this issue again. However, `dvc status` now does give me the expected answer without any Bad Request error message even though I currently have expired AWS credentials. Seems this issue was indeed fixed in 3.20 as suggested by @pmrowla, but I just needed to clear the cache?!
